python/pyarrow/_flight.pyx

```diff
@@ -844,8 +856,27 @@ cdef class FlightStreamReader(MetadataRecordBatchReader):
             (<CFlightStreamReader*> self.reader.get()).Cancel()
 
 
-cdef class FlightStreamWriter(_CRecordBatchWriter):
-    """A RecordBatchWriter that also allows writing application metadata."""
+cdef class MetadataRecordBatchWriter(_CRecordBatchWriter):
+    """A RecordBatchWriter that also allows writing application metadata.
+
+    This class is a context manager; on exit, close() will be called.
```

Comment:
Is `close` a method in the base class? If so, how about defining `__enter__` and `__exit__` in the base class instead?

Reply:
Turns out this was already in the base class, so I removed the definition here.

Thanks for the review!